test/drenv/minikube.py

```diff
@@ -104,6 +104,103 @@ def ssh(profile, command):
     _watch("ssh", command, profile=profile)
 
 
+def setup_files():
+    """
+    Set up minikube to work with drenv. Must be called before starting the
+    first cluster.
+
+    To load the configuration you must call load_files() after a cluster is
+    created.
+    """
+    _setup_sysctl()
+    _setup_systemd_resolved()
+
+
+def load_files(profile):
+    """
+    Load configuration done in setup_files() before the minikube cluster was
+    started.
+
+    Must be called after the cluster is started, before running any addon. Not
+    need when starting a stopped cluster.
+    """
+    _load_sysctl(profile)
```

Comment:
So this works and is totally fine, however, it might be a tiny bit limiting;
I think it's very likely that new developers on this project will have varying hacking requirements to get minikube (or minikube+other operators in my case) working properly.

Would you consider executing user-placed scripts from `~/.drenv/minikube_setup`?

Reply:
Running same scripts for every cluster is too broad in scope. The way to run custom code is to add an addon in test/addons/myaddon/start and put your addon in the first of of one of the workers in the clusters you want to run on.

We don't have yet mechanism to run some addon before all other addons in a cluster, but we can add something like this, maybe:

```yaml
profiles:
  - name: my-profile
    before:
      # Run before all other addons
      - name: my-before-addon
    run:
      # Run in parallel                                                                                                                                                             
      - addons:
        - name: kubevirt
      - addons:
        - name: cdi 
    after:
      # Run after all other addons
      - name: my-after-addon
```

If you have a use case for running your own script for every drenv environment please open an issue describing the use case.